go/cli/mcap/cmd/attachments.go

```diff
@@ -13,25 +13,33 @@ import (
 	"github.com/spf13/cobra"
 )
 
-func printAttachments(w io.Writer, attachmentIndexes []*mcap.AttachmentIndex) {
+func printAttachments(w io.Writer, r io.Reader) error {
 	tw := tablewriter.NewWriter(w)
-	rows := make([][]string, 0, len(attachmentIndexes))
+	rows := make([][]string, 0)
 	rows = append(rows, []string{
 		"log time",
 		"name",
 		"content type",
 		"content length",
-		"offset",
 	})
-	for _, idx := range attachmentIndexes {
-		row := []string{
-			fmt.Sprintf("%d", idx.LogTime),
-			idx.Name,
-			idx.ContentType,
-			fmt.Sprintf("%d", idx.DataSize),
-			fmt.Sprintf("%d", idx.Offset),
-		}
-		rows = append(rows, row)
+
+	cbReader, err := mcap.NewCallbackReader(mcap.ReadOptions{
+		OnAttachment: func(a *mcap.Attachment) error {
+			row := []string{
+				fmt.Sprintf("%d", a.LogTime),
+				a.Name,
+				a.ContentType,
+				fmt.Sprintf("%d", len(a.Data)),
+			}
+			rows = append(rows, row)
+			return nil
+		},
+	})
+	if err != nil {
+		return err
+	}
+	if err := cbReader.Read(r); err != nil {
+		return err
```

Comment:
The CLI function for listing attachments shouldn't actually visit the attachments in the file. It should just read the attachment indexes from the summary section and get this info from there.

We will need support at some point to actually retrieve an attachment from the file, but these summary functions should only interact with the index - that's particularly important when reading a remote file.

Reply:
The trouble with this is that mcaps with no summary still exist and need to have accurate information reported. Maybe we should only do the full scan if we encounter no summary.